internal/core/workers/runtime_watcher_worker/runtime_watcher_worker.go

```diff
@@ -24,29 +24,84 @@ package runtime_watcher_worker
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
 	"github.com/topfreegames/maestro/internal/core/workers"
+	"go.uber.org/zap"
 )
 
 var _ workers.Worker = (*runtimeWatcherWorker)(nil)
 
 // runtimeWatcherWorker is a work that will watch for changes on the Runtime
 // and apply to keep the Maestro state up-to-date. It is not expected to perform
 // any action over the game rooms or the scheduler.
-type runtimeWatcherWorker struct{}
+type runtimeWatcherWorker struct {
+	scheduler   *entities.Scheduler
+	roomManager *room_manager.RoomManager
+	// TODO(gabrielcorado): should we access the port directly? do we need to
```

Comment:
@gabrielcorado  I didn't get your point here

Reply:
Until now, we're only accessing ports from core services. The workers are not services, so should we add this "interface" to some of our services?